script.js

```diff
@@ -1,4 +1,300 @@
+// The different game modes or turns
+var GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST = "CHOOSE HIGHEST_OR_LOWEST";
+var GAME_MODE_CHOOSE_NUMBER_OF_DICE = "CHOOSE_NUMBER_OF_DICE";
+var GAME_MODE_PLAYER_ONE_ROLL = "PLAYER_ONE_ROLL";
+var GAME_MODE_PLAYER_ONE_CHOOSE = "PLAYER_ONE_CHOOSE";
+var GAME_MODE_PLAYER_TWO_ROLL = "PLAYER_TWO_ROLL";
+var GAME_MODE_PLAYER_TWO_CHOOSE = "PLAYER_TWO_CHOOSE";
+
+// Capture user input
+var userInputs = [];
+var userInput = null;
+var gameMode = GAME_MODE_CHOOSE_HIGHEST_OR_LOWEST;
+
+// Arrays and variables to capture user dice rolls and number results
+var player1DiceRolls = [];
+var player1Number = 0;
+var player1NumberList = [];
+var player1Score = 0;
+var player2DiceRolls = [];
+var player2Number = 0;
+var player2NumberList = [];
+var player2Score = 0;
+var numberOfDice = 0;
+var numberOfDices = [];
+var diceIndex = 0;
+var inputIndex = 0;
+var numberIndex = 0;
+var numberOfDiceIndex = 0;
+
+// Dice Roll function
+var rollDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var chooseHighestModeOrLowestMode = function () {
+  // Decide how the game will auto-generate combined number depending on the mode chosen.
+  // User input validation
+  if (userInput != "highest" && userInput != "lowest") {
+    return `Please enter either "highest" to play the 'highest combined number mode' or "lowest" to play the 'lowest combined number mode'.`;
+  }
+
+  userInputs.push(userInput);
+
+  if (userInputs[inputIndex] == "highest") {
+    var outputText = `highest`;
+  } else if ((userInputs[inputIndex] = "lowest")) {
+    var outputText = `lowest`;
+  }
+
+  outputText =
+    `Welcome Player 1 and Player 2. <br> You have chosen the ` +
+    outputText +
+    ` combined number mode. <br> Please enter the number of dice you would like to play with.`;
+
+  // Move the game forward
+  gameMode = GAME_MODE_CHOOSE_NUMBER_OF_DICE;
+
+  return outputText;
+};
+
+var chooseNumberOfDice = function () {
+  // User input validation
+  if (!Number(userInput)) {
+    return `Please enter a valid number of dice to play with.`;
+  }
+
+  numberOfDices.push(userInput);
+
+  outputText = `Player 1 and Player 2, You have chosen to play with ${numberOfDices[numberOfDiceIndex]}. <br> <br> Player 1, Press 'Submit' to roll the dices.`;
+
+  // Move the game forward
+  gameMode = GAME_MODE_PLAYER_ONE_ROLL;
+
+  return outputText;
+};
+
+// Player 1's turn to roll dice
+var playerOneRoll = function () {
+  // Roll the dice
+  var counter = 0;
+  player1DiceRolls = [];
+  while (counter < numberOfDices[numberOfDiceIndex]) {
+    // Add dice result to an array
+    player1DiceRolls.push(rollDice());
+    counter += 1;
+  }
```

Comment:
I like the use of comments to make the loop more readable as well as the use of '+=' and array that you have learnt from class